### ISSUE TITLE
[Transforms] Leading trivia of first statement in constructor is before parameter properties.

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#7896]
**Expected**:

```js
var Event = (function () {
    function Event() {
        // TODO: remove
        this._listeners = [];
        this._listeners = [];
    }
```

**Actual**:

```js
var Event = (function () {
    function Event() {
        this._listeners = [];
        // TODO: remove
        this._listeners = [];
    }
```

Failed tests:
*  tests/cases/compiler/lambdaArgCrash.ts

**Comment by rbuckton** [comment]
The original source was (trimmed for brevity):

```ts
class Event {
  private _listeners = [];
  constructor() {
    // TODO: remove

    this._listeners = [];
  }
}
```

Note the whitespace between the comment and the assignment. The issue was with the fact that the generated statement lists for the final function didn't have any position information, so it could not emit the detached comment at the top of the constructor.

**Comment by rbuckton** [comment]
Fixed in #7951.
